src/measure_theory/function/strongly_measurable.lean

```diff
@@ -181,6 +181,70 @@ protected lemma tendsto_approx {m : measurable_space α} (hf : strongly_measurab
   ∀ x, tendsto (λ n, hf.approx n x) at_top (𝓝 (f x)) :=
 hf.some_spec
 
+/-- Similar to `strongly_measurable.approx`, but enforces that the norm of every function in the
+sequence is less than `c` everywhere. If `f` has norm a.e. bounded by `c`, this sequence of simple
+functions verifies `∀ᵐ x ∂μ, tendsto (λ n, hf.approx_bounded n x) at_top (𝓝 (f x))`. -/
+noncomputable
+def approx_bounded {m : measurable_space α}
+  [has_norm β] [has_smul ℝ β] (hf : strongly_measurable f) (c : ℝ) :
+  ℕ → simple_func α β :=
+λ n, (hf.approx n).map (λ x, (min 1 (c / ∥x∥)) • x)
+
+lemma tendsto_approx_bounded {β} {f : α → β} [normed_group β] [normed_space ℝ β]
+  {m m0 : measurable_space α} {μ : measure α}
+  (hf : strongly_measurable[m] f) {c : ℝ}
+  (hf_bound : ∀ᵐ x ∂μ, ∥f x∥ ≤ c) :
+  ∀ᵐ x ∂μ, tendsto (λ n, hf.approx_bounded c n x) at_top (𝓝 (f x)) :=
```

Comment:
You write it as an almost everywhere statement, but isn't the core of the argument a pointwise result, that at a point x with ||f x|| \le c, then hf.aprox_bounded c n x tends to f x? It could be worth having this as a separate lemma.